### PR TITLE
GDB-6752 Implement Create cluster page

### PR DESCRIPTION
**Pull request by yordanalexandrov**
- implemented a new page for creating a cluster
 - removed old visualization and conde
 - implemented panels with updatable cluster status information
 - disabled old cluster management tests, to be rewriten

### CODE REVIEW

File: src/js/angular/clustermanagement/controllers.js [hunk 1]
```diff
@@ -12,765 +12,184 @@ const modules = [
...
-function CloneRepositoryCtrl($scope, $modalInstance, clone) {
-
-    $scope.clone = angular.copy(clone);
-    $scope.ok = function () {
-        const patt = new RegExp('^[a-zA-Z0-9-_]+$');
-        $scope.isInvalidRepoName = !patt.test($scope.clone.repositoryID);
-        if (!$scope.isInvalidRepoName) {
-            $modalInstance.close($scope.clone);
+    $scope.addNodeToList = function (nodeRpcAddress) {
+        if (!nodeRpcAddress || $scope.clusterConfiguration.nodes.includes(nodeRpcAddress)) {
```
Comment (svilenvelikov) [inline src/js/angular/clustermanagement/controllers.js:185]: Shouldn't there be some notification in this case?

File: src/pages/cluster-management/cluster-configuration.html [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        <form id="cluster-configuration-form" class="form-horizontal" name="clusterConfigurationForm" novalidate="novalidate">
+            <fieldset>
+                <div class="row">
+                    <div class="col-lg-6">
+                        <!-- Simply to satisfy the HTML specs, because it's reported as bug from SonarCloud -->
+                        <legend style="display: none">{{'cluster_management.cluster_page.create_form.legend' | translate}}</legend>
+                        <div class="form-group row">
+                            <label for="election-minimum-timeout" class="col-lg-6 col-xl-5 col-form-label">{{'cluster_management.cluster_configuration_properties.election_min_timeout' | translate}}*</label>
+                            <div class="col-lg-6 col-xl-7" tooltip="{{'cluster_management.cluster_configuration_properties.election_min_timeout_tooltip' | translate}}">
+                                <input class="form-control" placeholder="This field is required" type="number" ng-model="clusterConfiguration.electionMinTimeout"
```
Comment (svilenvelikov) [inline src/pages/cluster-management/cluster-configuration.html:17]: Translate the placeholders

File: src/pages/cluster-management/cluster-configuration.html [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                                        tooltip-placement="top">
+                                    <span class="icon-plus"></span>
+                                    {{'cluster_management.cluster_page.add_node_btn' | translate}}
+                                </button>
+                            </div>
+                        </div>
+                        <div class="row">
+                            <div class="col-xs-12">
+                                <div ng-if="clusterConfiguration.nodes.includes(rpcAddress)" class="nodeAlreadyAdded alert alert-danger mt-1">
+                                    {{'cluster_management.cluster_page.errors.node_in_list' | translate}}
```
Comment (svilenvelikov) [inline src/pages/cluster-management/cluster-configuration.html:73]: `nodeAlreadyAdded` why camelcase and not dash separated?

File: src/pages/cluster-management/cluster-configuration.html [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+                                <h5>{{'cluster_management.cluster_page.cluster_nodes_list' | translate}}</h5>
+                                <div class="row">
+                                    <div class="col-xs-12">
+                                        <div ng-if="clusterConfiguration.nodes.length < 2" class="notEnoughNodesError alert alert-danger mt-1">
+                                            {{'cluster_management.cluster_page.errors.at_least_two_nodes' | translate}}
+                                        </div>
+                                    </div>
+                                </div>
+                                <ul ng-if="clusterConfiguration.nodes && clusterConfiguration.nodes.length" class="list-group nodesList">
+                                    <li ng-repeat="node in clusterConfiguration.nodes" class="list-group-item list-group-item-action d-flex justify-content-between align-items-center node">
```
Comment (svilenvelikov) [inline src/pages/cluster-management/cluster-configuration.html:88]: trackBy

File: src/pages/cluster-management/cluster-configuration.html [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+                                    {{'cluster_management.cluster_page.errors.preconditions_failed.label' | translate}}
+                                    <ul>
+                                        <li>{{'cluster_management.cluster_page.errors.preconditions_failed.valid_license' | translate}}</li>
+                                        <li>{{'cluster_management.cluster_page.errors.preconditions_failed.authentication_secret' | translate}}</li>
+                                        <li>{{'cluster_management.cluster_page.errors.preconditions_failed.nodes_reachable' | translate}}</li>
+                                        <li>{{'cluster_management.cluster_page.errors.preconditions_failed.no_cluster_configured' | translate}}</li>
+                                    </ul>
+                                </div>
+                                <div class="col-xs-6">
+                                    Nodes status
```
Comment (svilenvelikov) [inline src/pages/cluster-management/cluster-configuration.html:120]: translate

File: src/pages/cluster-management/clusterInfo.html [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+</div>
+
+<div class="ng-scope" ng-hide="clusterConfiguration">
+    <p class="alert alert-danger">
+        {{'cluster_management.no_cluster_configured' | translate}}
+    </p>
+</div>
+
+<div ng-if="clusterConfiguration">
+    <div class="row" ng-if="clusterConfiguration">
```
Comment (svilenvelikov) [inline src/pages/cluster-management/clusterInfo.html:36]: Do you need the double `clusterConfiguration` check?

File: src/pages/cluster-management/clusterInfo.html [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+                                    <dt>{{'cluster_management.cluster_configuration_properties.message_size_kb' | translate}}</dt>
+                                    <dd>{{clusterConfiguration.messageSizeKB}}</dd>
+                                    <dt>{{'cluster_management.cluster_configuration_properties.verification_timeout' | translate}}</dt>
+                                    <dd>{{clusterConfiguration.verificationTimeout}}</dd>
+                                </dl>
+                            </div>
+                            <div class="col-xs-4">
+                                <h4>{{'cluster_management.cluster_configuration.nodes_list_label' | translate}}</h4>
+                                <ul>
+                                    <li ng-repeat="node in clusterConfiguration.nodes">
```
Comment (svilenvelikov) [inline src/pages/cluster-management/clusterInfo.html:69]: trackBy?

File: src/js/angular/clustermanagement/controllers.js (hunk 1 above)
Comment (yordanalexandrov) [reply src/js/angular/clustermanagement/controllers.js:185]: There is a visual notification under the box that is shown when node is already in the list - `class="nodeAlreadyAdded". This is just a check in the event callback

File: src/pages/cluster-management/cluster-configuration.html (hunk 4 above)
Comment (yordanalexandrov) [reply src/pages/cluster-management/cluster-configuration.html:88]: It is just a list of strings for now. Will add trackBy if/when it becomes an object[]

File: src/pages/cluster-management/clusterInfo.html (hunk 6 above)
Comment (yordanalexandrov) [reply src/pages/cluster-management/clusterInfo.html]: No, removed it.

File: src/pages/cluster-management/clusterInfo.html (hunk 7 above)
Comment (yordanalexandrov) [reply src/pages/cluster-management/clusterInfo.html]: Just strings

File: src/pages/cluster-management/cluster-configuration.html (hunk 3 above)
Comment (yordanalexandrov) [reply src/pages/cluster-management/cluster-configuration.html:73]: I saw it like this and continued. I made them dash separated.

**Review by svilenvelikov — APPROVED** [review]

**Comment by yordanalexandrov** [comment]
Jenkins retry build

**Pull request merged**